### PR TITLE
maintain: automatically add grant when cluster added

### LINKED ISSUE #3258: When connecting a kubernetes cluster, the user should be given `cluster-admin` access

### PR DESCRIPTION
**Pull request by BruceMacD**
- when an admin adds a cluster to infra through the UI automatically grant them cluster-admin access on finish

## Summary
It is confusing that someone must grant themselves access to a cluster they just added to Infra. This automatically gives a user "cluster-admin" access upon adding a cluster.

## Checklist

- [x] Considered security implications of the change
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged

## Related Issues

Resolves #3258

### CODE REVIEW

File: ui/pages/destinations/add.js [hunk 1]
```diff
@@ -238,11 +255,12 @@ export default function DestinationsAdd() {
...
             </div>
           )}
-          <Link href='/destinations'>
-            <a className='flex-none items-center self-center rounded-md border border-transparent bg-black px-4 py-2 text-2xs font-medium text-white shadow-sm hover:bg-gray-800'>
-              Finish
-            </a>
-          </Link>
+          <button
+            className='inline-flex items-center rounded-md border border-transparent bg-black px-4 py-2 text-2xs font-medium text-white shadow-sm hover:cursor-pointer hover:bg-gray-800 disabled:cursor-not-allowed disabled:opacity-30'
+            onClick={finish}
```
Comment (hoyyeva) [inline ui/pages/destinations/add.js:260]: Oh! this is such a good idea!!
nit: we don't really need a finish function. we can put all that in `onClick={async() => { ... }}`

Comment (hoyyeva) [reply ui/pages/destinations/add.js:260]: and we need a type in button tag

Comment (BruceMacD) [reply ui/pages/destinations/add.js]: thanks for the feedback, made these changes

File: ui/pages/destinations/add.js [hunk 2]
```diff
@@ -238,11 +238,28 @@ export default function DestinationsAdd() {
...
-          <Link href='/destinations'>
-            <a className='flex-none items-center self-center rounded-md border border-transparent bg-black px-4 py-2 text-2xs font-medium text-white shadow-sm hover:bg-gray-800'>
-              Finish
-            </a>
-          </Link>
+          <button
+            className='inline-flex items-center rounded-md border border-transparent bg-black px-4 py-2 text-2xs font-medium text-white shadow-sm hover:cursor-pointer hover:bg-gray-800 disabled:cursor-not-allowed disabled:opacity-30'
+            type='button'
+            onClick={async () => {
+              // grant the person that added this cluster 'cluster-admin' access automatically
```
Comment (mxyng) [inline ui/pages/destinations/add.js:245]: I don't think it's safe to assume the individual adding the cluster should automatically get any grants, particularly `cluster-admin`.  It's possible this elevates their privileges beyond what they already had. I suggest not merging this change.

Comment (BruceMacD) [reply ui/pages/destinations/add.js:245]: If they're an admin in Infra then they could just grant themselves cluster admin access after adding the cluster, so this is not a privilege escalation.

**Review by hoyyeva — APPROVED** [review]

**Pull request merged**